mitodl/mit-open
# Adds Posthog support to the frontend.

### What are the relevant tickets?

Closes #589 

### Description (What does it do?)

Adds PostHog integration to the frontend with these settings:
- The timeout is configurable in the `.env` file
- The PostHog provider is pre-loaded with the feature flags the Django app knows about
- Enabling the provider can also be set in the `.env` file.

The PostHog provider is situated near the top of the provider stack for the app - right after `StrictMode` - which should allow for feature flag checking and event collection for all parts of the app.

### How can this be tested?

Automated tests should pass.

Test setup of the PostHog provider: If the `POSTHOG_ENABLED` flag is set to `True`, you should see the `PostHogProvider` in the Components tree in the React dev tools. You should not see it if the flag is set to False. 

Test pre-loading of flags: With the provider enabled, you should be able to see the flags that the Django app knows about pre-loaded into the provider by navigating to the provider component in React dev tools and looking through the options. The preloading will use the currently logged-in user's username as the unique user ID for retrieving flags. Doing it this way should also load the flags from cache (see PR https://github.com/mitodl/mit-open/pull/682) if there's a cached value.
